components/application-broker/internal/broker/provision.go

```diff
@@ -74,25 +74,45 @@ func (svc *ProvisionService) Provision(ctx context.Context, osbCtx osbContext, r
 	defer svc.mu.Unlock()
 
 	iID := internal.InstanceID(req.InstanceID)
-	paramHash := jsonhash.HashS(req.Parameters)
+	provisioningParams := &internal.ProvisioningParameters{
+		Parameters: req.Parameters,
+	}
 
 	switch state, err := svc.instanceStateGetter.IsProvisioned(iID); true {
 	case err != nil:
 		return nil, &osb.HTTPStatusCodeError{StatusCode: http.StatusBadRequest, ErrorMessage: strPtr(fmt.Sprintf("while checking if instance is already provisioned: %v", err))}
 	case state:
-		if err := svc.compareProvisioningParameters(iID, paramHash); err != nil {
+		conflictOccured, err := svc.hasInstanceDifferentProvisiongParameters(iID, provisioningParams)
```

Comment:
We should not check if it is `not found` error, cause its problem of our inconsistent state that we know that sth is provisioned, but we do not have that instance in our storage. 
In case of the OSB API: 
> 400 Bad Request	MUST be returned if the request is malformed or missing mandatory data

so we cannot use the 400 error. We should simply return in such case the Internal Error

```go
conflictOccured, err := svc.hasInstanceDifferentProvisiongParameters(iID, provisioningParams)
if err != nil { // error handling ALWAYS FIRST
  return nil, &osb.HTTPStatusCodeError{StatusCode: http.StatusInternalServerError, ErrorMessage: strPtr(fmt.Sprintf("while comparing provisioning parameters %v: %v", req.Parameters, err))}
}
if conflictOccured {
  return nil, &osb.HTTPStatusCodeError{StatusCode: http.StatusConflict, ErrorMessage: strPtr(fmt.Sprintf("while comparing provisioning parameters %v: %v", req.Parameters, err))}
}
```

Reply:
I agree, to avoid having dead code I just removed the whole logic and added proper information and possible follow-up in PR description